### ISSUE TITLE
Template and forward-compatibility

### ISSUE THREAD

**Issue by vasilisp** [ocsigen/ocsigen-start#372]
I just spent about 4 hours updating an app that I had last touched in September. 4 hours is not that much, but:it would be much longer if I weren't familiar with Eliom and weren't following Ocsigen Start development. Note that I had already spent a similar amount of time in September to fix similar API breakage (since July or August).

This brings up the very important issue of template and lib compatibility. Will the 1.0 template be compatible with the 2.0 library (with only minor changes)? If not, Ocsigen Start 2.0 will understandably frustrate users that adopt Ocsigen Start now.

To me, this looks like a fundamental issue of the lib+template model. It could be fixed with a very strict regime about the library API breakage. Are we prepared to commit to the 1.0 APIs, i.e., only update them in a way that retains compatibility with old templates? That sounds hard to achieve, given that the lib API is deeply tied to the UI behaviors of the template.

I consider this to be the number-1 issue with Ocsigen Start. (If you are curious, my number-2 would be the dependencies on dubious quality non-OCaml stuff.)

**Comment by balat** [comment]
I agree with this. We must be very careful with the library interface. Let's keep this in mind ...

We have dependencies only to very common external packages (postgresql, sass ...).
